aineko/core/dataset.py

```diff
@@ -381,30 +396,30 @@ def _delivery_report(err: Any, message: Message) -> None:
         if err is not None:
             logger.error("Message %s delivery failed: %s", message, err)
 
-    def produce(self, message: dict, key: Optional[str] = None) -> None:
+    def produce(
+        self, message: Union[dict, str], key: Optional[str] = None
+    ) -> None:
         """Produce a message to the dataset.
 
         Args:
             message: message to produce to the dataset
             key: key to use for the message
         """
-        message = {
-            "timestamp": datetime.datetime.now().strftime(
-                AINEKO_CONFIG.get("MSG_TIMESTAMP_FORMAT")
-            ),
-            "dataset": self.dataset,
-            "source_pipeline": self.source_pipeline,
-            "source_node": self.source_node,
-            "message": message,
-        }
+        wrapped_message = MessageData(
+            dataset=self.dataset,
+            message=message,
+            source_node=self.source_node,
+            source_pipeline=self.source_pipeline,
+        ).model_dump(mode="json")
+
         self.producer.poll(0)
 
         key_bytes = str(key).encode("utf-8") if key is not None else None
 
         self.producer.produce(
```

Comment:
My main concern here was ensuring this output looked consistent with what we've built so far and it looks good 👍 

Example: 

```
wrapped_message = MessageData(
    dataset="logging",
    message="Hello World!",
    source_node="logging_node",
    source_pipeline="logging_pipeline",
)
```

produces

`b'{"message": "Hello World!", "timestamp": "2024-01-25 11:22:25.463134", "dataset": "logging", "source_pipeline": "logging_pipeline", "source_node": "logging_node"}'
`

Reply:
So there shouldn't be any issues with compatibility